profit/sur/gp.py

```diff
@@ -209,67 +210,66 @@ def add_training_data(self, x, y, sigma=None):
         self.ytrain = np.concatenate([self.ytrain, y], axis=0)
         self.m.set_XY(X=self.xtrain, Y=self.ytrain)
 
-    def predict(self, x=None):
+    def predict(self, x=None, return_input=False):
         """ Predict output from unseen data with the trained model.
         :return: (mean, var)
         """
-        if x is not None:
-            self.xpred = x if x.ndim > 1 else x.reshape(-1, 1)
-        else:
-            minval = 0.8 * self.xtrain.min(axis=0)
-            maxval = 1.2 * self.xtrain.max(axis=0)
-            step = 0.01 * (maxval - minval) / maxval
-            npoints = ((maxval - minval) // step + 1).astype(int)
-            self.xpred = np.full((max(npoints), self.ndim), np.nan)
-            for d in range(self.ndim):
-                self.xpred[:npoints[d], d] = np.linspace(minval[d], maxval[d], npoints[d])
+        if x is None:
+            minval = self.xtrain.min(axis=0)
+            maxval = self.xtrain.max(axis=0)
+            npoints = [50] * len(minval)
+            x = [np.linspace(minv, maxv, n) for minv, maxv, n in zip(minval, maxval, npoints)]
+        xgrid = np.meshgrid(*x) if isinstance(x, list) else np.meshgrid([x[:, d] for d in range(x.shape[-1])])
```

Comment:
This becomes dangerous for high dimensions - maybe one should add a limit of dimension <= 2 or <= 3 and otherwise require input of x?